undirect.go

```diff
@@ -0,0 +1,158 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package graph
+
+// Undirect converts a directed graph to an undirected graph, resolving
+// edge weight conflicts.
+type Undirect struct {
+	G Directed
+
+	// Absent if the value used to
+	// represent absent edge weights
+	// passed to Merge if the reverse
+	// edge is present.
+	Absent float64
+
+	// Merge defines how discordant edge
+	// weights in G are resolved. A merge
+	// is performed if at least one edge
+	// exists between the nodes being
+	// considered.
+	// The order of weight parameters
+	// passed to Merge is not defined, so
+	// the function should be commutative.
+	// If Merge is nil, the arithmetic
+	// mean is used to merge weights.
+	Merge func(x, y float64) float64
```

Comment:
I think it is fine to pass the edges, they can be easily ignored if not necessary.